api/src/routes/identity/resolution/get.ts

```diff
@@ -30,7 +32,20 @@ export async function get(config: IConfiguration, request: IIdentityDidResolveRe
     const providerUrl = networkConfig.provider;
     const permanodeUrl = networkConfig.permaNodeEndpoint;
 
-    return resolveIdentity(request.did, providerUrl, permanodeUrl);
+    const identityResult = await resolveIdentity(request.did, providerUrl, permanodeUrl);
+
+    if (identityResult.error !== "DIDNotFound") {
+        return Promise.resolve(identityResult);
```

Comment:
Why the need for all the Promise.resolve() calls ?

Reply:
closely related to the return question above: the route function is expected to return a promise, we need to inspect the result of `resolveIdentity` to determine the identity version, that is why we can't return `resolveIdentity` as is. So we wrap the result again before returning to "promisify" the return. Does that make sense?

Reply:
But it is already an `async` method so will automatically return everything as a promise, no need for the additional wrapping with `Promise.resolve`

But there still must be a type rather than `Promise<unknown>`, surely it can return `Promise<IIdentityDidResolveResponse>`

Reply:
absolutely! Don't know what I was thinking :/